### PR TITLE
ID-1: From URL and multipart fixes

### PR DESCRIPTION
**Pull request by dmitrijivanchenko**
### [Documents conversion](https://uploadcare-evrone.atlassian.net/browse/UC-1)

- Add a client for each thread for multipart upload.
- Fix form data building for multipart and URL upload.

## Checklist

- [ ] Tests (if applicable)
- [ ] Documentation (if applicable)
- [ ] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

**Review by dmitry-mukhin — APPROVED** [review]

File: lib/uploadcare/client/multipart_upload/chunks_client.rb [hunk 1]
```diff
@@ -18,9 +18,10 @@ class ChunksClient < ApiStruct::Client
         # @param links [Array] of strings; by default list of Amazon storage urls
         def upload_chunks(object, links)
```
Comment (evgeniy-r) [inline lib/uploadcare/client/multipart_upload/chunks_client.rb:19]: I don't like that when we call `client.upload_chunks`, we actually don't use `client` in any way.
Can we make this method class-level?

Comment (evgeniy-r) [reply lib/uploadcare/client/multipart_upload/chunks_client.rb:19]: And than maybe add
```ruby
        def upload_chunks(object, links)
          self.class.upload_chunks(object, links)
        end
```

File: lib/uploadcare/client/uploader_client.rb [hunk 2]
```diff
@@ -74,19 +74,35 @@ def get_status_response(token)
       def upload_many_body(arr, **options)
         files_formdata = arr.map do |file|
           [HTTP::FormData::File.new(file).filename,
-           HTTP::FormData::File.new(file)]
+           form_data_for(file)]
         end.to_h
+        p Param::Upload::UploadParamsGenerator.call(options[:store]).merge(files_formdata)
```
Comment (evgeniy-r) [inline lib/uploadcare/client/uploader_client.rb:79]: Should we remove this?

File: lib/uploadcare/client/uploader_client.rb [hunk 3]
```diff
@@ -74,19 +74,35 @@ def get_status_response(token)
...
         HTTP::FormData::Multipart.new(
           Param::Upload::UploadParamsGenerator.call(options[:store]).merge(files_formdata)
         )
       end
 
+      def form_data_for(file)
+        filename = file.original_filename if file.respond_to?(:original_filename)
+        mime_type = file.content_type if file.respond_to?(:content_type)
+        options = { filename: filename, content_type: mime_type }.compact
+        HTTP::FormData::File.new(file, options)
```
Comment (evgeniy-r) [inline lib/uploadcare/client/uploader_client.rb:89]: Duplication with `MultipartUploaderClient#multiupload_metadata`.

File: lib/uploadcare/client/uploader_client.rb [hunk 4]
```diff
@@ -74,19 +74,35 @@ def get_status_response(token)
...
+      end
+
+      STORE_VALUES = {
```
Comment (evgeniy-r) [inline lib/uploadcare/client/uploader_client.rb:92]: `STORE_VALUES_MAP`?

Comment (evgeniy-r) [reply lib/uploadcare/client/uploader_client.rb:92]: Or we can define a method for the store value...

File: lib/uploadcare/client/multipart_upload/chunks_client.rb [hunk 5]
```diff
@@ -16,11 +16,11 @@ class ChunksClient < ApiStruct::Client
         # In multiple threads, split file into chunks and upload those chunks into respective Amazon links
         # @param object [File]
         # @param links [Array] of strings; by default list of Amazon storage urls
-        def upload_chunks(object, links)
+        def self.upload_chunks(object, links)
           Parallel.each(0...links.count, in_threads: Uploadcare.config.upload_threads) do |link_id|
             offset = link_id * CHUNK_SIZE
             chunk = IO.read(object, CHUNK_SIZE, offset)
-            upload_chunk(chunk, links[link_id])
+            new.send(:upload_chunk, chunk, links[link_id])
```
Comment (evgeniy-r) [inline lib/uploadcare/client/multipart_upload/chunks_client.rb:23]:
```suggestion
            new.upload_chunk(chunk, links[link_id])
```

**Pull request merged**